Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/RideExtra.hs

```diff
@@ -613,52 +613,87 @@ findTotalRidesInDay (Id driverId) time = do
-  rideIds <- findIdsByFleetOwner fleetOwnerId
+totalRidesByFleetOwnerPerDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> Id Person -> UTCTime -> UTCTime -> m Int
+totalRidesByFleetOwnerPerDriver fleetOwnerId driverId from to = do
+  rideIds <- findIdsByFleetOwner fleetOwnerId from to
   getCompletedRidesByDriver (cast <$> rideIds) driverId
 
-totalRidesByFleetOwnerPerVehicleAndDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> Text -> Id Person -> m Int
-totalRidesByFleetOwnerPerVehicleAndDriver fleetOwnerId vehicleNumber driverId = do
-  rideIds <- findIdsByFleetOwnerAndVehicle fleetOwnerId vehicleNumber
+totalRidesByFleetOwnerPerVehicleAndDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> Text -> Id Person -> UTCTime -> UTCTime -> m Int
+totalRidesByFleetOwnerPerVehicleAndDriver fleetOwnerId vehicleNumber driverId from to = do
+  rideIds <- findIdsByFleetOwnerAndVehicle fleetOwnerId vehicleNumber from to
   getCompletedRidesByDriver (cast <$> rideIds) driverId
 
-totalEarningsByFleetOwnerPerVehicleAndDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> Text -> Id Person -> m Int
-totalEarningsByFleetOwnerPerVehicleAndDriver fleetOwnerId vehicleNumber driverId = do
-  rideIds <- findIdsByFleetOwnerAndVehicle fleetOwnerId vehicleNumber
+totalEarningsByFleetOwnerPerVehicleAndDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> Text -> Id Person -> UTCTime -> UTCTime -> m Int
+totalEarningsByFleetOwnerPerVehicleAndDriver fleetOwnerId vehicleNumber driverId from to = do
+  rideIds <- findIdsByFleetOwnerAndVehicle fleetOwnerId vehicleNumber from to
   getEarningsByDriver (cast <$> rideIds) driverId
+
+totalRidesStatsInFleet :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Maybe Text -> UTCTime -> UTCTime -> m (Int, Int, Int, Int)
+totalRidesStatsInFleet fleetOwnerId from to = do
+  rideIds <- findIdsByFleetOwner fleetOwnerId from to
+  (totalEarning, distanceTravelled, totalCompletedRides) <- getCompletedRidesStatsByIds (cast <$> rideIds)
+  totalCancelledRides <- getRidesByIdAndStatus (cast <$> rideIds) DRide.CANCELLED
+  pure (totalEarning, distanceTravelled, totalCompletedRides, totalCancelledRides)
+
+fleetStatsByDriver :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => [Id RideDetails.RideDetails] -> Id Person -> m (Int, Int, Int, (Int, Int))
+fleetStatsByDriver rideIds driverId = do
+  (totalEarning, distanceTravelled, totalCompletedRides, duration) <- getCompletedRidesStatsByIdsAndDriverId (cast <$> rideIds) driverId
+  let (hours, minutes) = calculateTimeDifference duration
+  pure (totalEarning, distanceTravelled, totalCompletedRides, (hours, minutes))
+
+fleetStatsByVehicle :: (CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => Text -> Text -> UTCTime -> UTCTime -> m (Int, Int, Int, (Int, Int))
+fleetStatsByVehicle fleetOwnerId vehicleNumber from to = do
```

Comment:
Using tuple of tuple is not a good style. For future more clear types should be used in these functions